### PR TITLE
Guide: Querying

### PR DESCRIPTION
**Pull request by bobbilee19**

### CODE REVIEW

**Review by angiejones — CHANGES_REQUESTED** [review]
please write this as a guide. don't want to clutter the API doc. can show one example on api doc and link to guide for more info and examples

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+# Query from DWN
+
+Before reading from your Decentralized Web Node (DWN), be sure to check out our [Quickstart guide](../../quickstart) to ensure you’ve properly imported the Web5 SDK. Additionally, make sure you’ve set up a DID before attempting to query from a DWN.
+
+## Querying Protocols
+
+The following snippet allows you to query for protocols within your `Web5` instance’s DWN:
+
+```jsx
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:13]: js

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const { protocols } = await web5.dwn.protocols.query({
+  message: {
+    filter: {
+      protocol: "https://music.org/protocol",
+    },
+  },
+});
+
+// logs an array of protocol configurations installed on the user's DWN
+console.log(protocols);
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:23]: can remove this and the comment above. not necessary

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+sidebar_position: 6
+---
+
+# Query from DWN
+
+Before reading from your Decentralized Web Node (DWN), be sure to check out our [Quickstart guide](../../quickstart) to ensure you’ve properly imported the Web5 SDK. Additionally, make sure you’ve set up a DID before attempting to query from a DWN.
+
+## Querying Protocols
+
+The following snippet allows you to query for protocols within your `Web5` instance’s DWN:
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:11]: - change "your" to "a"
- remove "Web5 instance's"

> The following snippet queries a DWN for protocols that match a given schema

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    },
+  },
+});
+
+// logs an array of protocol configurations installed on the user's DWN
+console.log(protocols);
+```
+
+## Querying Records
+
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:27]: Need a sentence explaining what this code snippet does

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  message: {
+    filter: {
+      schema: 'https://schema.org/Playlist',
+      dataFormat: 'application/json'
+    }
+  }
+});
+
+/// logs an array of records found on the user's DWN
+console.log(records);
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:40]: remove lines 39-40

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+## Query Sorting
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:43]: Sorting Query Results

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:44]: Add a sentence explaining that query results can be sorted by including the `dateSort` field. Then list the valid values for date sort and explain what each does. 

- createdAscending - sorts results by dateCreated in ascending order
- createdDescending
- publishedAscending
- publishedDescending

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+### Records
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:45]: Ascending Order

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/api/web5-js/dwn/protocols.mdx [hunk 9]
```diff
@@ -329,4 +329,18 @@ const { protocols } = await web5.dwn.protocols.query({
 // logs an array of protocol configurations installed on Bob's DWN
 console.log(protocols);
 ```
+
+```js
+// Sorting protocols by dateCreated in descending order
+const { protocols } = await web5.dwn.protocols.query({
+  message: {
+    filter: {
+      protocol: 'http://social-mediasss.xyz',
```
[the remark this answers is not in the document]
Comment (bobbilee19) [reply site/api/web5-js/dwn/protocols.mdx:338]: hahah it's funny i was typing on my keyboard and i couldn't see where my cursor was and never found the edit.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 1 above)
Comment (bobbilee19) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:13]: yep

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 4 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:24]: hey, I don't see the sentence added

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 7 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:40]: can you write it in a bulleted list so that it's easy to read?
image

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 4 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:24]: Can say:

> The following snippet queries a DWN for records that match a given schema and are in a specific data format.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+      schema: 'https://schema.org/Playlist',
+      dataFormat: 'application/json'
+    }
+  }
+});
+```
+
+## Sorting Query Results
+
+Query results can also be sorted via the dateSort field.
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:39]: put `dateSort` in code format

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+The valid values for dateSort are:
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:41]: put `dateSort` in code format

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 7 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:40]: I addressed this

image

**Review by angiejones — APPROVED** [review]

**Pull request merged**